### PR TITLE
Cu 1x793hc future proof rating view

### PR DESCRIPTION
**Pull request by Matthew5002**
Still things to discuss in terms of server side and typings to support this. Also, still needs to be connected to database.

### CODE REVIEW

File: app.config.ts [hunk 1]
```diff
@@ -31,8 +31,7 @@ export default {
       favicon: './assets/favicon.png'
     },
     extra: {
-      serverAddress: console.error('serverAddress must be specified.')
-      
+      serverAddress: 'http://be0e-195-213-145-15.ngrok.io'
```
Comment (bridges-wood) [inline app.config.ts]: Could you please revert this to the previous version?

File: src/App.tsx [hunk 2]
```diff
@@ -16,12 +29,13 @@ const client = new ApolloClient({
...
 const App = () => {
-
   return (
     <ApolloProvider client={client}>
       <View style={styles.container}>
         {/* <MapView/> */}
-        <SearchView/>
+        {/* <SearchView/> */}
+        {/*<BusinessView businessToDisplay={ExampleBusiness}/>*/}
+        <FutureProofRatingView businessToDisplay={ExampleBusiness}/>
```
Comment (bridges-wood) [inline src/App.tsx:38]: Is this component ready to be reverted to the MapView?

File: src/components/BusinessView.tsx [hunk 3]
```diff
@@ -1,44 +1,73 @@
...
 import React from 'react'
-import { SafeAreaView, ScrollView, StyleSheet } from 'react-native'
-import ImagesCarousel from './ImagesCarousel'
+import { SafeAreaView, ScrollView, StyleSheet, TouchableOpacity, Text, View } from 'react-native'
+import openMap from 'react-native-open-maps'
+import ImagesCarousel from '@components/ImagesCarousel'
 
-// TODO - Update to fetch graphQL
-const ExampleBusiness : DisplayableBusiness =  {
+// PLACEHOLDER BUSINESS LOCATION -> Update to fetch graphQL
```
Comment (bridges-wood) [inline src/components/BusinessView.tsx:11]: Are we ready to transition yet?

File: src/components/BusinessView.tsx [hunk 4]
```diff
@@ -1,44 +1,73 @@
...
+    backgroundColor: '#1EA853',
+    borderRadius: 10,
+    bottom: 40,
+    justifyContent: 'center',
+    padding: 18,
+    position: 'absolute',
+    width: '90%',
+    zIndex: 0
+  },
+  DirectionsButtonText: {
```
Comment (bridges-wood) [inline src/components/BusinessView.tsx]: Could you revert to the current style convention of names with a lowercase first letter?

File: src/components/BusinessViewMap.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  id: '1',
+  address: 'Unit B Strahan House, Avon St, Bath BA1 1UN',
+  latitude: 51.37758520597919,
+  longitude: -2.3572347659685513
+}
+
+it('matches snapshot', () => {
+  const { toJSON } = render(<BusinessViewMap businessLocation={ExampleBusinessLocation}/>)
+  expect(toJSON()).toMatchSnapshot()
+})
```
Comment (bridges-wood) [inline src/components/BusinessViewMap.spec.tsx:16]: Was this the only test you could think of?

File: src/components/RatingView.spec.tsx [hunk 6]
```diff
@@ -6,15 +6,3 @@ it('matches snapshot', () => {
...
-  const { getByTestId } = render(<RatingView ratingName='FutureProof' ratingValue={null}/>)
-  const text = getByTestId('rating-value')
-  expect(text).toHaveTextContent('0')
-})
-
-it('sets the ratingValue to 0 if it is undefined', () => {
-  const { getByTestId } = render(<RatingView ratingName='FutureProof' ratingValue={undefined}/>)
-  const text = getByTestId('rating-value')
-  expect(text).toHaveTextContent('0')
-})
```
Comment (bridges-wood) [inline src/components/RatingView.spec.tsx:20]: Why were these tests removed?

File: src/components/RatingView.tsx [hunk 7]
```diff
@@ -10,8 +11,7 @@ interface ComponentProps {
 const RatingView = ({ ratingName, ratingValue = 0 } : ComponentProps ) => (
   <View style={styles.itemBackgroundStyle}>
     <View style={styles.ratingContainerStyle}>
-      <Text>{`${ratingName} Rating`}</Text>
-      <Text testID='rating-value'>{ratingValue == null ? 0 : ratingValue}</Text>
+      <CircularRatingIndicator circleWidth={150} circleHeight={150} progressBarWidth={14} progressValue={ratingValue == null ? 0 : ratingValue} ratingName={ratingName}/>
```
Comment (bridges-wood) [inline src/components/RatingView.tsx:14]: Do you think it would be a good idea to pass the null value through and do the check there rather than cluttering this component?

File: src/components/SearchResultSorter.tsx [hunk 8]
```diff
@@ -1,7 +1,7 @@
 import BottomSheet from '@components/BottomSheet'
 import Button from '@components/Button'
 import SortOptions from '@components/SortOptions'
-import { SortOptions as SortOptionsType } from '@futureproof/typings'
+import { SortOptions as  SortOptionsType } from '@futureproof/typings'
```
Comment (bridges-wood) [inline src/components/SearchResultSorter.tsx]: Can you revert this?

File: src/components/TitleView.tsx [hunk 9]
```diff
@@ -3,12 +3,12 @@ import React from 'react'
 import { Image, StyleSheet, Text, View } from 'react-native'
 
 type ViewProps = Pick<DisplayableBusiness, 'name' | 'profilePicture'>
-  
-const TitleView = ({ name, profilePicture } : ViewProps ) => (
+
+const TitleView = ({ name, profilePicture, businessAddress } : {name : string, profilePicture : string, businessAddress : string} ) => (
```
Comment (bridges-wood) [inline src/components/TitleView.tsx]: Why did you change from using ViewProps?

File: src/components/ratings/CertificateLabel.tsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    right: 5,
+    width: 2
+  },
+  CertificateTitleStyle: {
+    fontSize: 10
+  },
+  IndicatorBackgroundStyle: {
+    alignItems: 'center',
+    flexDirection: 'row'
+  }
```
Comment (bridges-wood) [inline src/components/ratings/CertificateLabel.tsx]: Please revert to the previous naming convention here too. Lowercase first letter.

File: src/components/ratings/CertificatesList.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,8 @@
+import CertificatesList from '@components/ratings/CertificatesList'
+import { render } from '@testing-library/react-native'
+import React from 'react'
+
+it('matches snapshot', () => {
+  const { toJSON } = render(<CertificatesList/>)
+  expect(toJSON()).toMatchSnapshot()
+})
```
Comment (bridges-wood) [inline src/components/ratings/CertificatesList.spec.tsx]: Are there no more possible tests here?

File: src/components/ratings/CertificatesList.tsx [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+export const styles = StyleSheet.create({
+  CertificateItemStyle: {
+    paddingVertical: 5
+  },
+  CertificatesListStyle: {
+    alignSelf: 'center'
+  }
+})
```
Comment (bridges-wood) [inline src/components/ratings/CertificatesList.tsx]: See previous style comments. Lowercase first letter

File: src/components/ratings/CircularRatingIndicator.tsx [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+  RatingNameStyle: {
+    alignItems: 'center',
+    position: 'absolute',
+    zIndex: 1
+  },
+  RatingValueStyle: {
+    alignItems: 'center',
+    fontWeight: 'bold',
+    position: 'absolute',
+    zIndex: 1
```
Comment (bridges-wood) [inline src/components/ratings/CircularRatingIndicator.tsx]: Style names start with a lowercase letter

File: src/components/ratings/FutureProofRatingView.tsx [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+          <RectangularRatingIndicator progressValue={20} ratingName={'Working Conditions'}/>
+        </View>
+        <View style={styles.RectangularRatingStyle}>
+          <RectangularRatingIndicator progressValue={95} ratingName={'Ecosystem Impact'}/>
+        </View>
+        <View style={styles.RectangularRatingStyle}>
+          <RectangularRatingIndicator progressValue={51} ratingName={'Pay distibution'}/>
+        </View>
+        <View style={styles.RectangularRatingStyle}>
+          <RectangularRatingIndicator progressValue={85} ratingName={'Discrimination and Diversity'}/>
```
Comment (bridges-wood) [inline src/components/ratings/FutureProofRatingView.tsx]: Shouldn't we pull these values from the server - or has @CM20314-Group-2/backend not implemented these yet?

File: src/components/ratings/FutureProofRatingView.tsx [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+    marginHorizontal: 30
+  },
+  RectangularRatingStyle: {
+    paddingVertical: 5
+  },
+  headingText: {
+    color: '#A0A0A0',
+    fontSize: 12,
+    paddingLeft: 20
+  }
```
Comment (bridges-wood) [inline src/components/ratings/FutureProofRatingView.tsx]: Lowercase first letter please

File: src/components/ratings/RectangularRatingIndicator.spec.tsx [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+it('throws an error when progressValue is less than 0', () => {
+  expect(() => render(<RectangularRatingIndicator progressValue={-1} ratingName={'MyRating'} />)).toThrow()
+})
+
+it('throws an error when progressValue is greater than 100', () => {
+  expect(() => render(<RectangularRatingIndicator progressValue={101} ratingName={'MyRating'} />)).toThrow()
+})
+
+/*
+Don't currently know how to fetch SVG elements, replace below if found:
```
Comment (bridges-wood) [inline src/components/ratings/RectangularRatingIndicator.spec.tsx:26]: Did you try [this](https://testing-library.com/docs/queries/bytitle/)?

File: src/components/ratings/RectangularRatingIndicator.tsx [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+    justifyContent: 'space-evenly',
+    padding: 10
+  },
+  IndicatorStyle: {
+    borderRadius: 5,
+    height: 10
+  },
+  RatingTitleStyle: {
+    fontSize: 12
+  }
```
Comment (bridges-wood) [inline src/components/ratings/RectangularRatingIndicator.tsx]: Lowercase first letter.

**Review by bridges-wood — CHANGES_REQUESTED** [review]
Excellent overall. Most comments are about the naming convention for styles, but otherwise a stellar job.

File: app.config.ts (hunk 1 above)
Comment (Matthew5002) [reply app.config.ts]: Yes, reverting this back in next commit.

File: src/components/BusinessViewMap.spec.tsx (hunk 5 above)
Comment (Matthew5002) [reply src/components/BusinessViewMap.spec.tsx:16]: I'll add a few, missed this.

File: src/components/RatingView.spec.tsx (hunk 6 above)
Comment (Matthew5002) [reply src/components/RatingView.spec.tsx:20]: Because rating view was changed to use the new CircularRatingIndicator instead of the old text-based system.

File: src/components/RatingView.tsx (hunk 7 above)
Comment (Matthew5002) [reply src/components/RatingView.tsx:14]: Yes.

File: src/components/SearchResultSorter.tsx (hunk 8 above)
Comment (Matthew5002) [reply src/components/SearchResultSorter.tsx]: yes

File: src/components/TitleView.tsx (hunk 9 above)
Comment (Matthew5002) [reply src/components/TitleView.tsx]: ViewProps uses Displayable business which doesn't have the business address.

File: src/components/ratings/CertificatesList.spec.tsx (hunk 11 above)
Comment (Matthew5002) [reply src/components/ratings/CertificatesList.spec.tsx]: It's currently a static view because of waiting on server-side to implement the required datatypes.

File: src/components/ratings/FutureProofRatingView.tsx (hunk 14 above)
Comment (Matthew5002) [reply src/components/ratings/FutureProofRatingView.tsx]: I haven't seen this implemented server-side yet, that's why I've left it as a static view.

File: src/components/ratings/RectangularRatingIndicator.spec.tsx (hunk 16 above)
Comment (Matthew5002) [reply src/components/ratings/RectangularRatingIndicator.spec.tsx:26]: There doesn't appear to be a suitable way to do this. I've tried various ways with no success, and online suggests it's an issue without a solution currently.

File: src/App.tsx (hunk 2 above)
Comment (Matthew5002) [reply src/App.tsx:38]: Yes, just a placeholder to test + demonstrate how to use this view.

File: src/components/BusinessView.tsx (hunk 3 above)
Comment (Matthew5002) [reply src/components/BusinessView.tsx:11]: I haven't been involved with bridging the server and client that much so I was going to leave this to sprint 3 when Cam A's changes should integrate this view into the app passing the required data into the view to make server requests.

File: src/components/BusinessView.tsx (hunk 4 above)
Comment (Matthew5002) [reply src/components/BusinessView.tsx]: Done

File: src/components/BusinessViewMap.spec.tsx (hunk 5 above)
Comment (Matthew5002) [reply src/components/BusinessViewMap.spec.tsx:16]: I want to discuss this view because I wanted to be able to use the MapView for this view too to access the pins, etc by passing in certain parameters.

Comment (cg-2611) [reply src/components/BusinessViewMap.spec.tsx:16]: Im not too sure on how this component works or some of its functionality,  but other tests could include checking that certain styles are correct when certain events occur or that certain functions are called when certain buttons are pressed etc.

File: src/components/BusinessViewMap.tsx [hunk 18]
```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import { StyleSheet, Text, View } from 'react-native'
+import { ExampleLocationType } from '@components/BusinessView'
+import Map from 'react-native-maps'
+
+const BusinessViewMap = ({ businessLocation } : { businessLocation : ExampleLocationType }) => (
```
Comment (cg-2611) [inline src/components/BusinessViewMap.tsx:6]: I believe you might be able to use the MapView in the business profile and set the prop showRadius to false to hide the distance radius and just have the map and pins. We can always change what gets hidden in when this prop is false if you don't want the pins to be displayed too.

File: src/components/ratings/CertificatesList.spec.tsx (hunk 11 above)
Comment (cg-2611) [reply src/components/ratings/CertificatesList.spec.tsx]: you could test if the correct certificate name is rendered when it is passed to the component through a prop

File: src/components/ratings/CertificatesList.tsx [hunk 19]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import { StyleSheet, View } from 'react-native'
+import CertificateLabel from '@components/ratings/CertificateLabel'
+
+const CertificatesList = () => {
```
Comment (cg-2611) [inline src/components/ratings/CertificatesList.tsx]: I don't know about the certificates but you might be able to store each one in an array and use the js map function to create a certificate label for each certificate in the array. This way the component is more generalised and it is easier to add more certificates or remove certificates by adding or removing from the array.

File: src/components/ratings/CircularRatingIndicator.tsx [hunk 20]
```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import { StyleSheet, Text, View } from 'react-native'
+import Svg, { Circle } from 'react-native-svg'
+import { ratingToColour } from '@components/search/RatingCapsule'
+
+const BACKGROUND_SECONDARY_COLOUR = '#FAF9F9'
+
+const CircularRatingIndicator = ({ circleWidth, circleHeight, progressBarWidth, progressValue, ratingName } : { circleWidth : number, circleHeight : number, progressBarWidth : number, progressValue : number, ratingName : string }) => {
```
Comment (cg-2611) [inline src/components/ratings/CircularRatingIndicator.tsx:8]: extract typings to a ComponentProps interface

File: src/components/ratings/FutureProofRatingView.spec.tsx [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+it('matches snapshot', () => {
+  const { toJSON } = render(<FutureProofRatingView businessToDisplay={ExampleBusiness}/>)
+  expect(toJSON()).toMatchSnapshot()
+})
+
+it('renders rating title text appended with "Rating"', () => {
+  const { getByTestId } = render(<FutureProofRatingView businessToDisplay={ExampleBusiness} />)
+  const nameText = getByTestId('rating-name-text')
+  expect(nameText).toHaveTextContent('FutureProof Rating')
+})
```
Comment (cg-2611) [inline src/components/ratings/FutureProofRatingView.spec.tsx]: add a blank new line at end of file

File: src/components/ratings/FutureProofRatingView.tsx [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+
+const FutureProofRatingView = ({ businessToDisplay } : { businessToDisplay : DisplayableBusiness }) => {
+  return (
+    <SafeAreaView style={styles.futureProofRatingViewStyle}>
+      <ScrollView>
+        <View style={styles.futureProofRatingTitleView}>
+          <CircularRatingIndicator circleWidth={150} circleHeight={150} progressBarWidth={14} progressValue={businessToDisplay.sustainabilityScore ?? 0} ratingName={'FutureProof'}/>
+          <CertificatesList/>
+        </View>
+        <Text style={styles.headingText}>BREAKDOWN</Text>
```
Comment (cg-2611) [inline src/components/ratings/FutureProofRatingView.tsx:35]: This might also be able to be generalised to a map function applied to an array of objects containing a ratingName ad progressValue

**Review by cg-2611 — CHANGES_REQUESTED** [review]

**Review by cg-2611 — APPROVED** [review]

**Review by bridges-wood — APPROVED** [review]
Deferred to @cg-2611
